lib/irb/context.rb

```diff
@@ -151,6 +151,10 @@ def initialize(irb, workspace = nil, input_method = nil)
       @command_aliases = @user_aliases.merge(KEYWORD_ALIASES)
     end
 
+    private def term_interactive?
+      STDIN.tty? && ENV['TERM'] != 'dumb'
```

Comment:
How about `STDIN.tty? && STDOUT.tty?`? Does this solves your problem?

When STDIN and STDOUT are both tty, it is executed in an interactive terminal.
Reline and IRB both have a tty check, but there is an inconsistency. IRB only checks `STDIN.tty?`, and on the other hand, Reline only checks `$stdout.tty?`. If `$stdout` is not a tty or TERM is dumb, Reline enters unusable test mode.

`ENV['TERM'] == 'dumb'` is used for testing IRB with Reline. Although using `ENV['TERM']` like this is a bit weird, changing the TERM=dumb behavior right now will break the test.

Reply:
Unfortunately, both `STDIN.tty?` and `STDOUT.tty?` return true when the shell is running inside Emacs. But apparently some of the terminal features it's using don't work as expected in that environment. So treating "dumb" terminals differently makes sense to me (that's the definition - terminal with limited capabilities).

Do you know any other real-world "dumb" terminals? It could be useful to test IRB in one of them as well.

Reply:
The tests are indeed currently broken by this change, any recommendations for finding a fix would be welcome.

Reply:
I see. Thanks for the explanation. I agree IRB should treat dumb terminals differently :+1:

For test, could you add a way to test IRB by forcing `term_interactive?` to be true?
Let it return true if `ENV["TEST_IRB_FORCE_INTERACTIVE"]` is set, and in IRB's test which uses `PTY.spawn`, specify both TEST_IRB_FORCE_INTERACTIVE and TERM=dumb env.